agent/xds/xds_protocol_helpers_test.go

```diff
@@ -166,9 +166,6 @@ func newTestServerDeltaScenario(
 ) *testServerScenario {
 	mgr := newTestManager(t)
 	envoy := NewTestEnvoy(t, proxyID, token)
-	t.Cleanup(func() {
-		envoy.Close()
-	})
```

Comment:
cleanups and defers are run in LIFO order. I wanted to ensure `envoy.Close()` got called before the global metrics sink was reset